ci: enable test assets and cache them

Tachi107 opened the pull request:
As suggested in https://github.com/wwylele/teakra/pull/63#issuecomment-3160441721

Tachi107 committed:
ci: enable test assets and cache them

Tachi107 committed:
ci: pass --build-config to ctest

It's required when using multi-config generators like MSVC on Windows

Tachi107 committed:
ci: limit runs to the master branch

This avoids duplicated runs on merge requests

wwylele commented:
git-lfs sounds good to me. It is also good for archiving. I don't think I have used my quota at all

Tachi107 commented:
I cannot add git-lfs files to the project myself. The way I did so was:

```text
git lfs install
mkdir tests/data
touch tests/data/.keep
git add tests/data/.keep
git commit -m 'test: create tests/data directory'
mv build/teaklite2_tests_result tests/data/
git lfs track --filename tests/data/teaklite2_tests_result
git add .gitattributes
git commit -m 'test: track teaklite2_tests_result with git-lfs'
git add tests/data/teaklite2_tests_result
git commit -m 'test: add teaklite2_tests_result'
git lfs ls-files
baffcd4f80 * tests/data/teaklite2_tests_result
```

wwylele commented:
I have uploaded it to https://github.com/wwylele/teakra/tree/master/src/test_verifier/data. I realized this has been downloading from someone else' server in https://github.com/wwylele/teakra/blob/a0364b6319351d1900cbc862954265de9a35b2e1/src/test_verifier/CMakeLists.txt#L18. We should change that line and redirect it to the in-repo file, though I don't have set up to test this locally

Tachi107 committed:
Do not download git-lfs files by default

The test file is really big, and downloading it by default may create
issues with users on a slow network.

Use a .lfsconfig file to disable LFS downloads. This way, LFS files can
be explicitly downloaded via

    git lfs pull --exclude='' --include='*'

Tachi107 committed:
build: use Git LFS test data

Tachi107 commented:
Hi again! I've reworked with patch set, combining Git LFS with GitHub Actions caching. This way, multiple repeated runs will download the big test asset from the cache instead of LFS storage. The cache is (supposed to be) faster, and LFS bandwidth is limited.

I've also disabled Git LFS downloads by default, since most users are unlikely to want to download that test asset, and it takes a long time to do so

wwylele commented:
Thanks!
